code/src/LeafletProvider/DrawingTools/DrawingTools.ts

```
@@ -0,0 +1,240 @@
+        }
+
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        private _addCompletedEventHandler(event: any): void {
+            const toolType = event.layerType;
+
+            this.tools
+                .filter((element) => element.type === toolType)
+                .forEach((element) => element.addCompletedEvent(event));
+        }
+
+        private _getDrawingToolsPosition(
+            position: string
+        ): Constants.DrawingTools.Positions {
+            if (Constants.DrawingTools.Positions[position] !== undefined) {
+                return Constants.DrawingTools.Positions[position];
+            } else {
+                OSFramework.Helper.ThrowError(
+                    this.map,
+                    OSFramework.Enum.ErrorCodes.CFG_InvalidDrawingToolsPosition,
+                    `${position}`
+                );
+                return;
+            }
+        }
+
+        //TODO: create the return structure
+        private _getTools(): ToolsList {
+            const _tools = new ToolsList();
+            this.tools.forEach((tool) => {
+                switch (tool.type) {
+                    case OSFramework.Enum.DrawingToolsTypes.Circle:
+                        _tools.circle = tool.options;
+                        break;
+                    case OSFramework.Enum.DrawingToolsTypes.Marker:
+                        _tools.marker = tool.options;
+
+                        break;
+                    case OSFramework.Enum.DrawingToolsTypes.Polygon:
+                        _tools.polygon = tool.options;
+                        break;
+                    case OSFramework.Enum.DrawingToolsTypes.Polyline:
+                        _tools.polyline = tool.options;
+
+                        break;
+                    case OSFramework.Enum.DrawingToolsTypes.Rectangle:
+                        _tools.rectangle = tool.options;
+                        break;
+                    default:
+                        break;
```

Maybe throw error stating that this tool is not defined?